### PR TITLE
feat: add Xcode 14 guards, bump minimum iOS version to 13.0

### PR DESCRIPTION
**Pull request by hansemannn**
## General Setup & Build Support
- Remove outdated guards for Xcode 11 and Xcode 12 (oudated since 2-3 years)
- Bump minimum build target to iOS 13 (yearly bump)
- Remove outdated docs with "minver <= 12" usages
- Remove outdated iOS <= 13 guards in the ObjC source

## New APIs
- The new `severity` property for alert dialog to determine a default vs critical alert dialog severity (primarily used on macOS catalyst)
- New `autolink` constants for text areas

This can be merged after master has been bumped to 12.0.0!

### CODE REVIEW

File: apidoc/Titanium/Network/Network.yml [hunk 1]
```diff
@@ -568,15 +568,15 @@ properties:
     permission: read-only
     since: {android: "3.6.0", iphone: "1.8.0.1", ipad: "1.8.0.1"}
     platforms: [android, iphone, ipad, macos]
-    osver: {android: {min: "4.1"}, ios: {min: "5.0"}}
+    osver: {android: {min: "4.1"}}
```
Comment (m1ga) [inline apidoc/Titanium/Network/Network.yml:571]: think we can do the same for the Android part since min is Android 6 (23). But that can be a different PR but I like the idea to remove such old details :+1:

Comment (hansemannn) [reply apidoc/Titanium/Network/Network.yml:571]: Isn't it Android 5 so far?

Comment (m1ga) [reply apidoc/Titanium/Network/Network.yml:571]: my bad. Was looking at a different package.json :wink: But searching for: `android: {min: "2`, `android: {min: "3`, `android: {min: "4`, `android: {min: "5.0` reveals some stuff that can be removed! Will do that after this one to reduce merge conflicts

Comment (m1ga) [reply apidoc/Titanium/Network/Network.yml:571]: btw: https://github.com/kaushikgopal/why_bump_android_minsdk to see what features you'll gain (or are always available) when you raise the minSDK.

Comment (hansemannn) [reply apidoc/Titanium/Network/Network.yml:571]: Interesting! Yeah I thought about bumping to 6 as well, but for now the market share of 5 is still to relevant :(
